src/urllib3/util/url.py

```diff
@@ -128,9 +128,27 @@ def request_uri(self) -> str:
 
         return uri
 
+    @property
+    def authority(self) -> Optional[str]:
+        """
+        Network location including userinfo, host and port.
+
+        i.e.
+            userinfo@host:port
+        """
+        if self.netloc is None or self.auth is None:
```

Comment:
Since a lot of these properties are calculated properties can we use variables like `auth = self.auth` and `netloc = self.netloc`.